docker-php-source

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+set -e
+PHP_SRC_DIR=/usr/src/php
+
+usage() {
+	echo "usage: $0 COMMAND"
+	echo
+	echo "Manage php source tarball lifecycle."
+	echo
+	echo "Commands:"
+	echo "   extract  extract php source tarball into directory $PHP_SRC_DIR if not already done."
+	echo "   delete   delete extracted php source located into $PHP_SRC_DIR if not already done."
+	echo
+}
+
+case "$1" in
+	extract)
+		if [ -e "$PHP_SRC_DIR" -a ! -d "$PHP_SRC_DIR" ] ; then
+			echo >&2 "$PHP_SRC_DIR exists and is not a directory"
+			exit 1
+		fi
+		if [ ! -d "$PHP_SRC_DIR" ]; then
+			mkdir -p $PHP_SRC_DIR
+			tar -Jxf /usr/src/php.tar.xz -C "$PHP_SRC_DIR" --strip-components=1
+
+			if [ ! -f /usr/src/available-php-exts ]; then
+				find "$PHP_SRC_DIR"/ext \
+						-mindepth 2 -maxdepth 2 \
+						-type f -name 'config.m4' \
+					| cut -d/ -f6 | sort \
```

Comment:
The `-f6` here is making assumptions about the number of path components in `$PHP_SRC_DIR`.  I'd recommend this block be adjusted to something more like the following:
```bash
(
	cd "$dir/ext"
	find \
		-mindepth 1 \
		-maxdepth 1 \
		-type f \
		-name 'config.m4' \
	| xargs -n1 dirname \
	| xargs -n1 basename
) > /usr/src/php-...
```